lib/engines/bh.js

```diff
@@ -27,26 +29,29 @@ module.exports = function (opts) {
 
         filePath = U.fulfillName(name, opts.ext || this.ext, opts.source);
 
-        // check cache options
-        if (opts.force || options.forceExec || options.forceLoad) {
-            dropRequireCache(require, name);
+        // drop cache
+        if (opts.force) {
+            delete cache[filePath];
         }
 
-        try {
-            bhModule = require(filePath);
-        } catch (e) {
-            cb(e);
-            return;
+        if (!cache[filePath]) {
+            try {
+                dropRequireCache(require, filePath);
+                cache[filePath] = require(filePath);
```

Comment:
I'm not sure that I see what's happened here. I don't like usage of `dropRequireCache` because it looks like a hack. Better to use public node api here like `readFile` if we need to reload it in runtime. `require` should be used only for modules because of sync nature.

So I suggest to preload and inject bh code into fixtures and run it under `mock-fs` to store data right in spec files to decrease complexity of tests at all.